common/components/PrintableWallet/index.jsx

```diff
@@ -0,0 +1,246 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import PropTypes from 'prop-types';
+import translate from 'translations';
+import QRCode from 'qrcode';
+import { toDataUrl as makeIdenticon } from 'ethereum-blockies';
+
+import ethLogo from 'assets/images/logo-ethereum-1.png';
+import sidebarImg from 'assets/images/print-sidebar.png';
+import notesBg from 'assets/images/notes-bg.png';
+
+const walletWidth = 680;
+const walletHeight = 280;
+
+export default class PrintableWallet extends Component {
+  static propTypes = {
+    privateKey: PropTypes.string,
+    address: PropTypes.string
+  };
+
+  state = {
+    qrCodePkey: null,
+    qrCodeAddress: null
+  };
+
+  componentDidMount() {
+    // Start generating QR Codes as soon as component is ready
+    const opts = {
+      color: {
+        dark: '#000',
+        light: '#fff'
+      },
+      margin: 0,
+      errorCorrectionLevel: 'H'
+    };
+
+    QRCode.toDataURL(this.props.privateKey, opts, (err, url) => {
+      this.setState({ qrCodePkey: url });
+    });
+
+    QRCode.toDataURL(this.props.address, opts, (err, url) => {
+      this.setState({ qrCodeAddress: url });
+    });
+  }
+
+  print = () => {
+    const popup = window.open(
+      'about:blank',
+      'printWalletWindow',
+      `width=${walletWidth + 60},height=${walletHeight + 60},scrollbars=no`
+    );
+
+    // We'll save ourselves from re-rendering by just using a ref for the html
+    focus();
+    popup.document.open();
+    popup.document.write(`
+			<html>
+				<head>
+					<style>
+						* {
+							box-sizing: border-box;
+						}
+
+						body {
+							font-family: Lato, sans-serif;
+							font-size: 1rem;
+							line-height: 1.4;
+							margin: 0;
+						}
+					</style>
+					<script>
+						setTimeout(function() {
+							window.print();
+						}, 500);
+					</script>
+				</head>
+				<body>
+					${this._wallet.outerHTML}
```

Comment:
Looks good, I avoided that at first because of the word "server", I assumed it relied on some node specific stuff. Turns out it's completely browser friendly!